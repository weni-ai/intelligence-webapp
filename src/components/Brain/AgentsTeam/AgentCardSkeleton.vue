<template>
  <section class="agent-card-skeleton">
    <UnnnicSkeletonLoading
      tag="div"
      width="30%"
      height="22px"
    />

    <UnnnicSkeletonLoading
      tag="div"
      width="100%"
      height="44px"
    />

    <section class="agent-card-skeleton__skills">
      <UnnnicSkeletonLoading
        tag="div"
        width="100px"
        height="30px"
      />
      <UnnnicSkeletonLoading
        tag="div"
        width="150px"
        height="30px"
      />
    </section>

    <UnnnicSkeletonLoading
      tag="div"
      width="100%"
      height="38px"
    />
  </section>
</template>

<style lang="scss" scoped>
.agent-card-skeleton {
  display: flex;
  flex-direction: column;
  gap: $unnnic-spacing-xs;

  &__skills {
    width: 100%;
    display: flex;
    gap: $unnnic-spacing-nano;
    flex-wrap: wrap;
  }
}
</style>
