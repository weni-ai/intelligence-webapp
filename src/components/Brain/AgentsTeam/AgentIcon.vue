<template>
  <img
    v-if="icon && agentIcon"
    :src="agentIcon"
    class="content__icon"
    data-testid="agent-icon"
  />
</template>

<script setup>
import { computed } from 'vue';
import { agentIcons } from '@/utils/agentIconService';

const props = defineProps({
  icon: {
    type: String,
    default: '',
  },
});

const agentIcon = computed(() => {
  if (!props.icon) return;

  return agentIcons[props.icon] || null;
});
</script>
