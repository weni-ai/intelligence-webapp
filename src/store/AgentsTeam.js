import { defineStore } from 'pinia';
import { computed, reactive, ref } from 'vue';

import nexusaiAPI from '@/api/nexusaiAPI.js';
import { useAlertStore } from './Alert';
import agentIconService from '@/utils/agentIconService';

import i18n from '@/utils/plugins/i18n';

export const useAgentsTeamStore = defineStore('AgentsTeam', () => {
  const linkToCreateAgent = 'https://github.com/weni-ai/weni-cli';

  const alertStore = useAlertStore();

  const activeTeam = reactive({
    status: null,
    data: {
      manager: null,
      agents: [],
    },
  });

  const officialAgents = reactive({
    status: null,
    data: [],
  });

  const myAgents = reactive({
    status: null,
    data: [],
  });

  const allAgents = computed(() => {
    return {
      manager: activeTeam.data.manager,
      agents: [...officialAgents.data, ...myAgents.data],
    };
  });

  const isAgentsGalleryOpen = ref(false);

  async function loadActiveTeam() {
    try {
      activeTeam.status = 'loading';

      const { data } = await nexusaiAPI.router.agents_team.listActiveTeam();

      activeTeam.data = {
        manager: data.manager,
        agents: agentIconService.applyIconsToAgents(data.agents),
      };
      activeTeam.status = 'complete';
    } catch (error) {
      console.error('error', error);

      activeTeam.status = 'error';
    }
  }

  async function loadOfficialAgents({ search = '' } = {}) {
    try {
      officialAgents.status = 'loading';

      const { data } = await nexusaiAPI.router.agents_team.listOfficialAgents({
        search,
      });

      officialAgents.data = agentIconService.applyIconsToAgents(data);
      officialAgents.status = 'complete';
    } catch (error) {
      console.error('error', error);

      officialAgents.status = 'error';
    }
  }

  async function loadMyAgents({ search = '' } = {}) {
    try {
      myAgents.status = 'loading';

      const { data } = await nexusaiAPI.router.agents_team.listMyAgents({
        search,
      });

      myAgents.data = agentIconService.applyIconsToAgents(data);
      myAgents.status = 'complete';
    } catch (error) {
      console.error('error', error);

      myAgents.status = 'error';
    }
  }

  async function toggleAgentAssignment({ uuid, is_assigned }) {
    if (!uuid || typeof is_assigned !== 'boolean') {
      throw new Error('uuid and is_assigned are required');
    }

    try {
      const agent =
        officialAgents.data.find((agent) => agent.uuid === uuid) ||
        myAgents.data.find((agent) => agent.uuid === uuid);

      if (!agent) {
        throw new Error('Agent not found');
      }

      const { data } =
        await nexusaiAPI.router.agents_team.toggleAgentAssignment({
          agentUuid: agent?.uuid,
          is_assigned,
        });

      agent.assigned = data.assigned;

      if (is_assigned) {
        activeTeam.data.agents.push(agent);
      } else {
        activeTeam.data.agents = activeTeam.data.agents.filter(
          (agent) => agent.uuid !== uuid,
        );
      }

      alertStore.add({
        text: i18n.global.t(
          is_assigned
            ? 'router.agents_team.card.success_assign_alert'
            : 'router.agents_team.card.success_unassign_alert',
          {
            agent: agent.name,
          },
        ),
        type: 'success',
      });

      return {
        status: 'success',
      };
    } catch (error) {
      console.error('error', error);

      alertStore.add({
        text: i18n.global.t('router.agents_team.card.error_alert'),
        type: 'error',
      });

      return {
        status: 'error',
      };
    }
  }

  return {
    linkToCreateAgent,
    activeTeam,
    officialAgents,
    myAgents,
    allAgents,
    isAgentsGalleryOpen,
    loadActiveTeam,
    loadOfficialAgents,
    loadMyAgents,
    toggleAgentAssignment,
  };
});
