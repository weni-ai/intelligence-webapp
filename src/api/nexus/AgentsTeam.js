import { computed } from 'vue';
import request from '@/api/nexusaiRequest';
import { useProjectStore } from '@/store/Project';
import { cleanParams } from '@/utils/http';

const projectUuid = computed(() => useProjectStore().uuid);

export const AgentsTeam = {
  async listOfficialAgents({ search }) {
    const params = cleanParams({
      search,
    });
    const { data } = await request.$http.get(
      `api/agents/official/${projectUuid.value}`,
      {
        params,
      },
    );

    return {
      data: data.map(
        ({
          uuid,
          name,
          description,
          skills,
          assigned,
          credentials,
          is_official,
          slug,
        }) => ({
          uuid,
          name,
          description,
          skills,
          assigned,
          credentials,
          is_official,
          id: slug,
        }),
      ),
    };
  },

  async listMyAgents({ search }) {
    const params = cleanParams({
      search,
    });
    const { data } = await request.$http.get(
      `api/agents/my-agents/${projectUuid.value}`,
      {
        params,
      },
    );

    return {
      data: data.map(
        ({
          uuid,
          name,
          description,
          skills,
          assigned,
          credentials,
          is_official,
          slug,
        }) => ({
          uuid,
          name,
          description,
          skills,
          assigned,
          credentials,
          is_official,
          id: slug,
        }),
      ),
    };
  },

  async listActiveTeam() {
    const { data } = await request.$http.get(
      `api/agents/teams/${projectUuid.value}`,
    );

    const { manager, agents } = data;

    return {
      data: {
        manager: {
          id: manager.id || 'manager',
        },
        agents: agents.map(
          ({
            uuid,
            name,
            skills,
            id,
            description,
            credentials,
            is_official,
          }) => ({
            uuid,
            name,
            skills,
            id,
            description,
            credentials,
            is_official,
          }),
        ),
      },
    };
  },

  async toggleAgentAssignment({ agentUuid, is_assigned }) {
    const { data } = await request.$http.patch(
      `api/project/${projectUuid.value}/assign/${agentUuid}`,
      {
        assigned: is_assigned,
      },
      {
        hideGenericErrorAlert: true,
      },
    );

    return {
      data,
    };
  },
};
