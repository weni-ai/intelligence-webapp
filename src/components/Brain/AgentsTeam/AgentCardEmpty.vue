<template>
  <button
    class="agent-card-empty"
    data-testid="agent-card-empty"
    @click="createNewAgent"
  >
    <UnnnicIcon
      icon="add"
      size="md"
      scheme="neutral-clean"
      data-testid="agent-card-empty-icon"
    />

    <UnnnicIntelligenceText
      tag="p"
      family="secondary"
      size="body-gt"
      color="neutral-dark"
      data-testid="agent-card-empty-title"
    >
      {{ $t('router.agents_team.create_agent') }}
    </UnnnicIntelligenceText>
  </button>
</template>

<script setup>
import { useAgentsTeamStore } from '@/store/AgentsTeam';

const agentsTeamStore = useAgentsTeamStore();
function createNewAgent() {
  window.open(agentsTeamStore.linkToCreateAgent, '_blank');
}
</script>

<style lang="scss" scoped>
.agent-card-empty {
  border: none;
  background: none;

  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  height: 100%;
  width: 100%;

  cursor: pointer;
}
</style>
