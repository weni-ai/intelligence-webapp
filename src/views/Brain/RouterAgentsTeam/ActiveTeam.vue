<template>
  <section class="active-team">
    <UnnnicIntelligenceText
      tag="h2"
      family="secondary"
      size="body-lg"
      color="neutral-darkest"
      weight="bold"
      data-testid="title"
    >
      {{ $t('router.agents_team.active_team.title') }}
    </UnnnicIntelligenceText>

    <section
      v-if="isLoadingTeam || activeTeam.length"
      class="active-team__cards"
    >
      <template v-if="isLoadingTeam">
        <AssignAgentCard
          v-for="(_, index) in Array(3)"
          :key="index"
          loading
          data-testid="loading-card"
        />
      </template>

      <template v-else>
        <AssignAgentCard
          v-for="agent in activeTeam"
          :key="agent.uuid"
          :agent="agent"
          :assignment="false"
          data-testid="team-card"
        />
      </template>
    </section>

    <section
      v-if="!isLoadingTeam && activeTeam.length === 0"
      class="active-team__empty"
      data-testid="empty-state"
    >
      <UnnnicIntelligenceText
        tag="p"
        family="secondary"
        size="body-gt"
        color="neutral-dark"
        weight="bold"
      >
        {{ $t('router.agents_team.active_team.no_team') }}
      </UnnnicIntelligenceText>

      <section class="empty__description">
        <!-- eslint-disable-next-line vue/component-name-in-template-casing -->
        <i18n-t
          class="description__text"
          keypath="router.agents_team.active_team.no_team_description"
          tag="p"
        >
          <template #link>
            <a
              class="description__link"
              :href="agentsTeamStore.linkToCreateAgent"
              target="_blank"
              >{{ $t('router.agents_team.active_team.create_agent') }}</a
            >
          </template>
        </i18n-t>
      </section>
    </section>
  </section>
</template>

<script setup>
import { computed, onMounted } from 'vue';

import { useAgentsTeamStore } from '@/store/AgentsTeam';

import AssignAgentCard from '@/components/Brain/AgentsTeam/AssignAgentCard.vue';

const agentsTeamStore = useAgentsTeamStore();
const activeTeam = computed(
  () => agentsTeamStore.activeTeam.data?.agents || [],
);

const isLoadingTeam = computed(
  () => agentsTeamStore.activeTeam.status === 'loading',
);

onMounted(() => {
  agentsTeamStore.loadActiveTeam();
});
</script>

<style lang="scss" scoped>
.active-team {
  border-radius: $unnnic-border-radius-md;
  border: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;

  padding: $unnnic-spacing-sm;

  display: flex;
  flex-direction: column;
  gap: $unnnic-spacing-sm;

  &__empty {
    border-radius: $unnnic-border-radius-md;
    border: $unnnic-border-width-thinner dashed $unnnic-color-neutral-cleanest;

    padding: $unnnic-spacing-xl $unnnic-spacing-sm;

    display: flex;
    flex-direction: column;
    align-items: center;

    background-color: $unnnic-color-neutral-lightest;

    .empty__description {
      display: flex;
      align-items: center;
      gap: 2px;

      .description__text {
        font-family: $unnnic-font-family-secondary;
        font-size: $unnnic-font-size-body-md;
        color: $unnnic-color-neutral-cloudy;
      }

      .description__link {
        font-size: $unnnic-font-size-body-md;
        color: $unnnic-color-neutral-cloudy;
        text-decoration: underline;
      }
    }
  }

  &__cards {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap: $unnnic-spacing-sm;
  }
}
</style>
