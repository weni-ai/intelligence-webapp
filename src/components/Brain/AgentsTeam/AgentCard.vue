<template>
  <section
    data-testid="agent-card"
    :class="['agent-card', { 'agent-card--empty': empty }]"
  >
    <AgentCardSkeleton
      v-if="loading"
      data-testid="agent-card-skeleton"
    />

    <AgentCardEmpty
      v-else-if="empty"
      data-testid="agent-card-empty"
    />

    <section
      v-else
      class="agent-card__content"
    >
      <UnnnicIntelligenceText
        tag="p"
        family="secondary"
        size="body-gt"
        color="neutral-darkest"
        weight="bold"
        data-testid="title"
      >
        {{ title }}
      </UnnnicIntelligenceText>

      <UnnnicIntelligenceText
        v-if="description"
        tag="p"
        family="secondary"
        size="body-gt"
        color="neutral-cloudy"
        data-testid="description"
      >
        {{ description }}
      </UnnnicIntelligenceText>

      <section class="content__skills">
        <Skill
          v-for="skill in skills"
          :key="skill.name"
          :title="skill.name"
          :icon="skill.icon"
          data-testid="skill"
        />
      </section>
    </section>

    <UnnnicButton
      v-if="!loading && !empty && assignment"
      :class="[
        'agent-card__button',
        { 'agent-card__button--assigned': assigned },
      ]"
      :text="
        assigned
          ? $t('router.agents_team.card.assigned')
          : $t('router.agents_team.card.assign')
      "
      :type="assigned ? 'tertiary' : 'primary'"
      :iconLeft="assigned ? 'check' : ''"
      size="small"
      :loading="isAssigning"
      data-testid="assign-button"
      @click="toggleAgentAssignment"
    />
  </section>
</template>

<script setup>
import { ref } from 'vue';

import { useAgentsTeamStore } from '@/store/AgentsTeam';

import AgentCardSkeleton from './AgentCardSkeleton.vue';
import AgentCardEmpty from './AgentCardEmpty.vue';
import Skill from './Skill.vue';

defineEmits(['assign']);

const props = defineProps({
  loading: {
    type: Boolean,
    default: false,
  },
  title: {
    type: String,
    default: '',
  },
  description: {
    type: String,
    default: '',
  },
  /**
   * An array of skills where each skill is an object with the following properties:
   * @property {string} name - The name of the skill
   * @property {string} icon - The icon of the skill
   */
  skills: {
    type: Array,
    default: () => [],
    validator(skills) {
      return skills.every((skill) => 'name' in skill && 'icon' in skill);
    },
  },
  assignment: {
    type: Boolean,
    default: true,
  },
  assigned: {
    type: Boolean,
    default: false,
  },
  uuid: {
    type: String,
    default: '',
  },
  empty: {
    type: Boolean,
    default: false,
  },
});

const isAssigning = ref(false);
const agentsTeamStore = useAgentsTeamStore();

async function toggleAgentAssignment() {
  isAssigning.value = true;

  try {
    await agentsTeamStore.toggleAgentAssignment({
      uuid: props.uuid,
      is_assigned: !props.assigned,
    });
  } catch (error) {
    console.error(error);
  } finally {
    isAssigning.value = false;
  }
}
</script>

<style lang="scss" scoped>
.agent-card {
  border-radius: $unnnic-border-radius-md;
  border: $unnnic-border-width-thinner solid $unnnic-color-neutral-cleanest;

  padding: $unnnic-spacing-sm;

  display: grid;
  gap: $unnnic-spacing-sm;
  align-content: space-between;

  &--empty {
    display: flex;

    cursor: pointer;
  }

  &__content {
    display: flex;
    flex-direction: column;
    gap: $unnnic-spacing-xs;

    .content__skills {
      display: flex;
      gap: $unnnic-spacing-nano;
      flex-wrap: wrap;
    }
  }

  :deep(.unnnic-button).agent-card__button {
    &--assigned {
      color: $unnnic-color-weni-600;
    }

    [class*='icon'] {
      color: $unnnic-color-weni-600;
    }
  }
}
</style>
