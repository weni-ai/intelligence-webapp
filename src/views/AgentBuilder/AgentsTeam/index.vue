<template>
  <section class="agents-team">
    <AgentBuilderHeader
      title="Agents Team"
      subtitle="Assign agents to your team"
      withDivider
      actionsSize="lg"
    >
      <template #actions>
        <section class="agents-team__actions">
          <UnnnicButton
            type="secondary"
            iconLeft="add"
            @click="handleAgentsGallery"
          >
            {{ $t('router.agents_team.assign_agents') }}
          </UnnnicButton>

          <UnnnicButton
            type="primary"
            iconLeft="play_arrow"
            iconsFilled
            @click="handlePreview"
          >
            {{ $t('router.agents_team.preview') }}
          </UnnnicButton>
        </section>
      </template>
    </AgentBuilderHeader>

    <ActiveTeam />

    <AgentsGalleryModal />

    <PreviewDrawer v-model="isPreviewOpen" />
  </section>
</template>

<script setup>
import { ref } from 'vue';

import AgentBuilderHeader from '@/components/AgentBuilder/Header.vue';
import PreviewDrawer from '@/components/Brain/Preview/PreviewDrawer.vue';

import { useAgentsTeamStore } from '@/store/AgentsTeam';

import ActiveTeam from './ActiveTeam.vue';
import AgentsGalleryModal from './AgentsGalleryModal.vue';

const isPreviewOpen = ref(false);

const handleAgentsGallery = () => {
  useAgentsTeamStore().isAgentsGalleryOpen = true;
};

const handlePreview = () => {
  isPreviewOpen.value = true;
};
</script>

<style lang="scss" scoped>
section.agents-team {
  display: grid;
  grid-template-rows: auto 1fr;

  .agents-team__actions {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: $unnnic-spacing-sm;
  }
}
</style>
