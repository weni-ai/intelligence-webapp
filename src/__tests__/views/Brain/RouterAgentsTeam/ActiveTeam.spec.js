import { shallowMount } from '@vue/test-utils';
import { nextTick } from 'vue';

import { createTestingPinia } from '@pinia/testing';
import { useAgentsTeamStore } from '@/store/AgentsTeam';

import ActiveTeam from '@/views/Brain/RouterAgentsTeam/ActiveTeam.vue';

const pinia = createTestingPinia({
  initialState: {
    AgentsTeam: {
      activeTeam: {
        status: null,
        data: {
          agents: [],
        },
      },
      officialAgents: {
        status: null,
        data: [],
      },
      myAgents: {
        status: null,
        data: [],
      },
    },
  },
});

describe('ActiveTeam.vue', () => {
  let wrapper;
  const agentsTeamStore = useAgentsTeamStore();

  const loadingCards = () =>
    wrapper.findAllComponents('[data-testid="loading-card"]');
  const teamCards = () =>
    wrapper.findAllComponents('[data-testid="team-card"]');
  const emptyState = () => wrapper.find('[data-testid="empty-state"]');
  const assignAgentsLink = () =>
    wrapper.find('[data-testid="assign-agents-link"]');

  const mockAgents = [
    {
      uuid: 'uuid-1',
      name: 'Agent 1',
      description: 'Description 1',
      skills: [{ name: 'Skill 1', icon: 'ðŸ›’' }],
    },
    {
      uuid: 'uuid-2',
      name: 'Agent 2',
      description: 'Description 2',
      skills: [{ name: 'Skill 2', icon: 'ðŸ’¬' }],
    },
  ];

  beforeEach(() => {
    wrapper = shallowMount(ActiveTeam, {
      global: {
        plugins: [pinia],
        stubs: {
          'i18n-t': {
            template: '<div><slot name="assign_agents" /></div>',
          },
        },
      },
    });
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  describe('Component rendering', () => {
    it('should match snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should load active team when mounted', () => {
      expect(agentsTeamStore.loadActiveTeam).toHaveBeenCalledTimes(1);
    });
  });

  describe('Loading state', () => {
    it('should render loading cards when team is loading', async () => {
      agentsTeamStore.activeTeam.status = 'loading';
      await nextTick();

      expect(loadingCards().length).toBe(3);
    });

    it('should not render empty state when loading', async () => {
      agentsTeamStore.activeTeam.status = 'loading';
      agentsTeamStore.activeTeam.data.agents = [];
      await nextTick();

      expect(emptyState().exists()).toBe(false);
    });
  });

  describe('Active team state', () => {
    it('should render team cards when there are active agents', async () => {
      agentsTeamStore.activeTeam.status = 'complete';
      agentsTeamStore.activeTeam.data.agents = mockAgents;
      await nextTick();

      expect(teamCards().length).toBe(mockAgents.length);
    });

    it('should not render empty state when there are active agents', async () => {
      agentsTeamStore.activeTeam.status = 'complete';
      agentsTeamStore.activeTeam.data.agents = [mockAgents[0]];
      await nextTick();

      expect(emptyState().exists()).toBe(false);
    });
  });

  describe('Empty state', () => {
    it('should render empty state when there are no agents', async () => {
      agentsTeamStore.activeTeam.status = 'complete';
      agentsTeamStore.activeTeam.data.agents = [];
      await nextTick();

      expect(emptyState().exists()).toBe(true);
    });

    it('should open agents gallery when clicking on assign agents link', async () => {
      agentsTeamStore.activeTeam.status = 'complete';
      agentsTeamStore.activeTeam.data.agents = [];
      await nextTick();

      await assignAgentsLink().trigger('click');
      expect(agentsTeamStore.isAgentsGalleryOpen).toBe(true);
    });
  });
});
