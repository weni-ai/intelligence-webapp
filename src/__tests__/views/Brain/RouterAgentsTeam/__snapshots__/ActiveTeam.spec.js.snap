// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActiveTeam.vue > Component rendering > should match snapshot 1`] = `
<section
  class="active-team"
  data-v-ef307fb1=""
>
  <!--v-if-->
  <section
    class="active-team__empty"
    data-testid="empty-state"
    data-v-ef307fb1=""
  >
    <icon-stub
      class="empty__icon"
      clickable="false"
      data-v-ef307fb1=""
      filled="true"
      icon="workspaces"
      next="false"
      scheme="neutral-soft"
      size="xl"
    />
    <unnnic-intelligence-text-stub
      color="neutral-dark"
      data-v-ef307fb1=""
      family="secondary"
      lineheight="md"
      size="body-lg"
      tag="p"
      weight="bold"
    />
    <section
      class="empty__description"
      data-v-ef307fb1=""
    >
      <!-- This comment prevents from auto-capitalizing i18n-t to I18nT which would break the component -->
      <!-- eslint-disable-next-line vue/component-name-in-template-casing -->
      <div
        class="description__text"
        data-v-ef307fb1=""
        keypath="router.agents_team.active_team.no_team_description"
        tag="p"
      >
        
        <p
          class="description__assign-agents"
          data-testid="assign-agents-link"
          data-v-ef307fb1=""
        >
          Assign agents
        </p>
        
      </div>
    </section>
  </section>
</section>
`;
